### PR TITLE
[SYCL][Level Zero][USM] Add a test for read-only shared alloc

### PR DESCRIPTION
**Pull request by aelovikov-intel**
For https://github.com/intel/llvm/pull/7496.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
Failures from "/verify with":

SYCL :: KernelAndProgram/kernel-bundle-merge-options-env.cpp - known for quite some time, being tracked/addressed already.
SYCL :: Regression/commandlist/gpu.cpp - OpenCL, hence unrelated. Not sure if being worked on. Failed on Linux in this run, not sure if that matters. Also, I wasn't able to find this failure on any other PR.

**Comment by aelovikov-intel** [comment]
"verify with" failure on

SYCL :: KernelAndProgram/kernel-bundle-merge-options-env.cpp

is being tracked in internal bug report.

**Review by smaslov-intel — APPROVED** [review]

File: SYCL/USM/shared_read_only.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// REQUIRES: level_zero
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER  %t.out h 2> %t1.out; cat %t1.out %GPU_CHECK_PLACEHOLDER
```
Comment (againull) [inline SYCL/USM/shared_read_only.cpp]: NIT: It looks "h" is a typo in "%t.out h"

**Pull request merged**